pkg/security/auth.go

```diff
@@ -187,10 +187,28 @@ func UserAuthCertHook(
 		if ValidateUserScope(certUserScope, systemIdentity.Normalized(), tenantID) {
 			return nil
 		}
-		return errors.Errorf("requested user %s is not authorized for tenant %d", systemIdentity, tenantID)
+		return errors.WithDetailf(errors.Errorf("certificate authentication failed for user %q", systemIdentity),
+			"You are connecting to tenant %v. The client cert is valid for %s.", tenantID, FormatUserScopes(certUserScope))
 	}, nil
 }
 
+// FormatUserScopes formats a list of scopes in a human-readable way,
+// suitable for e.g. inclusion in error messages.
+func FormatUserScopes(certUserScope []CertificateUserScope) string {
+	var buf strings.Builder
+	comma := ""
+	for _, scope := range certUserScope {
+		fmt.Fprintf(&buf, "%s%q on ", comma, scope.Username)
+		if scope.Global {
+			buf.WriteString("all tenants")
+		} else {
+			fmt.Fprintf(&buf, "tenant %v", scope.TenantID)
+		}
+		comma = ", "
```

Comment:
[style] If we implemented String() on security.CertificateUserScope, then this could be 

```golang
		fmt.Fprintf(&buf, "%s%s", comma, scope.String())
		comma = ", "
```

which I suppose allocate a bit more, so I leave it to your judgement.